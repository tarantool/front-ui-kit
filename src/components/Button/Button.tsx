
import React, { ReactNode, MouseEvent, forwardRef, ComponentType } from 'react';
import { cx } from '@emotion/css';

import { IconSpinner } from '../Icon';
import { styles, intentStyles, intentActiveStyles } from './Button.styles';

export interface ButtonIconProps {
  className?: string;
}

export interface ButtonProps {
  autoFocus?: boolean,
  className?: string,
  children?: ReactNode,
  disabled?: boolean,
  icon?: ComponentType<ButtonIconProps>,
  iconRight?: ComponentType<ButtonIconProps>,
  intent?: 'primary' | 'secondary' | 'base' | 'plain' | 'dark',
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void,
  loading?: boolean,
  size?: 'l' | 's' | 'xs' | 'm',
  text?: string,
  title?: string,
  type?: 'button' | 'submit'
}

export const Button = forwardRef<HTMLButtonElement, ButtonProps>((
  {
    autoFocus,
    className,
    children,
    disabled,
    icon: Icon,
    iconRight: IconRight,
    intent = 'base',
    onClick,
    loading,
    size = 'm',
    text,
    title,
    type = 'button',
    ...props
  }: ButtonProps,
  ref
) => {
  const isOnlyIcon = Icon && !children && !text;

  const content: JSX.Element[] = [];

  if (Icon) {
    content.push(
      <Icon
        key="Icon"
        className={cx(
          styles.icon,
          {
            [styles.iconMargin]: !isOnlyIcon
          }
        )}
      />
    )
  }

  content.push(<React.Fragment key="content">{children || text}</React.Fragment>)

  if (IconRight && !isOnlyIcon) {
    content.push(
      <IconRight
        key="IconRight"
        className={cx(
          styles.icon,
          styles.iconRightMargin
        )}
      />
    )
  }

  if (loading && !disabled) {
    content.push(
      <div key="loading" className={styles.loadingWrap}>
        <IconSpinner />
      </div>
    );
  }

  return (
    <button
      {...props}
      autoFocus={autoFocus}
      className={cx(
        styles.button,
        {
          [styles.iconicL]: isOnlyIcon && size === 'l',
          [styles.iconicM]: isOnlyIcon && size === 'm',
          [styles.iconicS]: isOnlyIcon && size === 's',
          [styles.l]: !isOnlyIcon && size === 'l',
          [styles.m]: !isOnlyIcon && size === 'm',
          [styles.s]: !isOnlyIcon && size === 's',
          [styles.xs]: size === 'xs',
          [intentActiveStyles[intent]]: !loading && !disabled
        },
        intentStyles[intent],
        {
          [styles.loading]: loading && !disabled
        },
        className
      )}
      disabled={disabled}
      onClick={onClick}
      title={title}
      type={type}
      ref={ref}
    >
      {content}
    </button>
  );
});
