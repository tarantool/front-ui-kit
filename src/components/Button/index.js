// @flow
import * as React from 'react';
import { css, cx } from 'emotion'

const styles = {
  btn: css`
    border: solid 1px gray;
  `
}

type ButtonProps = {
  className?: string,
  children?: React.Node,
  onClick?: (MouseEvent) => void
};

export default ({ children, className, ...props }: ButtonProps) => (
  <button className={cx(className, styles.btn)} {...props}>{children}</button>
);
