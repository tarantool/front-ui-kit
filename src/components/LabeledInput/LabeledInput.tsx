import React, { FC } from 'react';
import { cx } from '@emotion/css';

import { nanoid } from '../../utils/nanoid';
import { ControlsPanel } from '../ControlsPanel';
import { commonFormFieldStyles as commonStyles } from '../FormField/commonStyles';
import { IconInfo } from '../Icon';
import type { InputProps } from '../Input';
import { Input } from '../Input';
import { Text } from '../Text';
import { Tooltip } from '../Tooltip';

import { styles } from './LabeledInput.styles';

type LabeledInputProps = InputProps & {
  inputComponent?: React.ComponentType<any>;
  inputClassName?: string;
  className?: string;
  id?: string;
  info?: string;
  label: string;
  largeMargins?: boolean;
  subTitle?: string | React.ReactNode;
  topRightControls?: React.ReactNode[];
  error?: boolean;
  message?: string;
  preserveMessageSpace?: boolean;
};

const internalId = nanoid();

export const LabeledInput: FC<LabeledInputProps> = ({
  inputComponent: InputComponent = Input,
  topRightControls,
  className,
  id = internalId,
  inputClassName,
  subTitle,
  info,
  label,
  largeMargins,
  error,
  message,
  preserveMessageSpace,
  ...props
}: LabeledInputProps) => {
  return (
    <div className={cx(commonStyles.wrap, { [commonStyles.wrapMargin]: largeMargins }, className)}>
      <div className={cx(commonStyles.headingPane, { [commonStyles.headingPaneMargin]: largeMargins })}>
        <Text className={commonStyles.label} variant="h4" tag="label">
          {label}
        </Text>
        {!!info && (
          <Tooltip className={commonStyles.tooltip} content={info}>
            <IconInfo />
          </Tooltip>
        )}
        {!!subTitle && (
          <Text className={commonStyles.subTitle} variant="p" tag="span">
            {subTitle}
          </Text>
        )}
        {topRightControls && <ControlsPanel className={commonStyles.topRightControls} controls={topRightControls} />}
      </div>
      <InputComponent {...props} error={error} id={id} className={cx(styles.input, inputClassName)} />
      {(preserveMessageSpace || message) && (
        <Text variant="p" className={cx(styles.message, { [styles.errorMessage]: error })}>
          {message}
        </Text>
      )}
    </div>
  );
};
