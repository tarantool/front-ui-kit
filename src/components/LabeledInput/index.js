// @flow
import * as React from 'react';
import { css, cx } from 'emotion';
import { Text } from '../Text';
import { Tooltip } from '../Tooltip';
import { IconInfo } from '../Icon';
import { ControlsPanel } from '../ControlsPanel';

const styles = {
  wrap: css`
    display: block;
    margin-bottom: 24px;
  `,
  tooltip: css`
    display: inline-block;
    margin-left: 8px;
  `,
  headingPane: css`
    display: flex;
    flex-direction: row;
    align-items: baseline;
    margin-bottom: 8px;
  `,
  subTitle: css`
    margin-left: 32px;
  `,
  label: css`
    display: block;
  `,
  topRightControls: css`
    margin-left: auto;
  `,
  errorMessage: css`
    display: block;
    height: 20px;
    color: #F5222D;
    position: absolute;
  `
};

type LabeledInputProps = {
  children?: React.Node,
  className?: string,
  info?: string,
  label: string,
  subTitle?: string | React.Node,
  topRightControls?: React.Node[],
  errorMessage?: string,
};

export const LabeledInput = ({
  children,
  topRightControls,
  className,
  subTitle,
  info,
  label,
  errorMessage
}:
LabeledInputProps) => (
  <label className={cx(styles.wrap, className)}>
    <div className={styles.headingPane}>
      <Text className={styles.label} variant='h4' tag='span'>{label}:
        {info && (
          <Tooltip className={styles.tooltip} content={info}>
            <IconInfo />
          </Tooltip>
        )}
      </Text>
      {subTitle && <Text className={styles.subTitle} variant='h5' tag='span' upperCase>{subTitle}</Text>}
      {topRightControls && <ControlsPanel className={styles.topRightControls} controls={topRightControls} />}
    </div>
    {children}
    <Text variant='p' className={styles.errorMessage}>{errorMessage}</Text>
  </label>
);
