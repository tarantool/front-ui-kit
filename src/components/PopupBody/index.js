// @flow

// TODO: move popup components inside Modal
import * as React from 'react';
import { css, cx } from '@emotion/css';
import { Scrollbar } from '../Scrollbar';

const styles = {
  wrap: css`
    width: 100%;
    height: 100%;
    padding-left: 30px;
    padding-right: 30px;
    overflow: hidden;
  `,
  scrollableWrap: css`
    height: 100%;
    margin-left: -30px;
    margin-right: -30px;
  `,
  scrollableBody: css`
    padding-left: 32px;
    padding-right: 32px;
  `
};

type PopupBodyProps = {
  children?: React.Node,
  className?: string,
  innerClassName?: string,
  scrollable?: boolean
};

export const PopupBody = ({ children, className, innerClassName, scrollable }: PopupBodyProps) => scrollable
  ? (
    <Scrollbar className={cx(styles.scrollableWrap, className)}>
      <div className={cx(styles.scrollableBody, innerClassName)}>
        {children}
      </div>
    </Scrollbar>
  )
  : (
    <div className={cx(styles.wrap, className)}>
      {children}
    </div>
  );
