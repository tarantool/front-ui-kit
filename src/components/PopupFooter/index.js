// @flow
import * as React from 'react';
import { css, cx } from 'emotion';
import { ControlsPanel } from '../ControlsPanel';

const styles = {
  wrap: css`
    display: flex;
    padding: 8px 16px;
  `,
  controls: css`
    padding-left: 16px;
    margin-left: auto;
  `
};

type PopupFooterProps = {
  children?: React.Node,
  className?: string,
  controls?: React.Node[]
};

export const PopupFooter = ({ children, className, controls }: PopupFooterProps) => (
  <div className={cx(styles.wrap, className)}>
    {children}
    {controls && (
      <ControlsPanel className={cx(styles.controls)} thin>
        {controls}
      </ControlsPanel>
    )}
  </div>
);
