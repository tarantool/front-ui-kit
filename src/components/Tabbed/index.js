// @flow
import * as React from 'react';
import { css, cx } from '@emotion/css';
import { textStyles } from '../Text';
import { colors, zIndex } from '../../variables';

const styles = {
  default: {
    wrap: css`
      display: flex;
      flex-direction: column;
      height: 100%;
    `,
    tab: css`
      position: relative;
      padding: 14px 0;
      border: none;
      margin-right: 30px;
      margin-bottom: -1px;
      border-bottom: solid 2px transparent;
      color: ${colors.dark40};
      background-color: transparent;
      outline: none;
      cursor: pointer;

      &:hover,
      &:focus {
        z-index: ${zIndex.inline};
        color: ${colors.dark65};
      }

      &:last-child {
        margin-right: 0;
      }
    `,
    activeTab: css`
      color: ${colors.dark};
      border-bottom-color: ${colors.dark};
    `,
    tabs: css`
      border-bottom: solid 1px ${colors.intentBase};
    `,
  },
  small: {
    wrap: css`
      display: flex;
      flex-direction: column;
    `,
    tab: css`
      position: relative;
      cursor: pointer;
      border: none;
      font-size: 14px;
      color: ${colors.dark};
      width: 100%;
      margin-left: 8px;
      margin-right: 8px;
      padding: 9px 20px;
      border-radius: 4px;
      background-color: ${colors.baseBg};

      &:focus {
        background-color: ${colors.baseBg};
      }

      &:hover {
        z-index: ${zIndex.inline};
        background-color: #fff;
      }
    `,
    activeTab: css`
      background-color: #fff !important;
      box-shadow: 0px 2px 2px rgba(4, 11, 29, 0.05);
    `,
    tabs: css`
      display: flex;
      justify-content: space-around;
      padding 7px 8px;
      background-color: ${colors.baseBg};
      border-radius: 4px;
    `,
  },
};

type Tab = {
  label: string,
  content: React.Node,
};

type TabbedProps = {
  tabs?: Tab[],
  className?: string,
  activeTab?: number,
  handleTabChange?: (i: number, e: MouseEvent) => void,
  size: 'default' | 'small',
};

type TabbedState = {
  activeTab: number,
};

export class Tabbed extends React.Component<TabbedProps, TabbedState> {
  static defaultProps = {
    size: 'default',
  };

  state = {
    activeTab: 0,
  };

  render() {
    const { className, tabs = [], activeTab: activeTabProps, size } = this.props;
    const { activeTab } = this.state;
    const tabIndex = activeTabProps ?? activeTab;

    return (
      <div className={cx(styles[size].wrap, className)}>
        <div className={styles[size].tabs}>
          {tabs &&
            tabs.map(({ label }, i) => (
              <button
                key={i}
                className={cx({ [textStyles.h3]: size === 'default' }, styles[size].tab, {
                  [styles[size].activeTab]: tabIndex === i,
                })}
                onClick={(e) => this.handleTabChange(i, e)}
              >
                {label}
              </button>
            ))}
        </div>
        {tabs[tabIndex].content}
      </div>
    );
  }

  handleTabChange(i: number, e: MouseEvent) {
    e.preventDefault();
    this.setState({ activeTab: i });
    this.props?.handleTabChange?.(i, e);
  }
}
