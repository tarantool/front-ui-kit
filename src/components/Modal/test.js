import React from 'react';
import renderer from 'react-test-renderer';
import { css } from 'emotion';
import { Modal } from './index';
import { Button } from '../Button';
import { Tabbed } from '../Tabbed';
import { PopupBody } from '../PopupBody';
import { PopupFooter } from '../PopupFooter';
import { LabeledInput } from '../LabeledInput';

jest.mock('react-dom', () => ({
  createPortal: node => node
}));

jest.mock(
  '../Scrollbar/index.js',
  () => ({
    Scrollbar: ({ children, className }) => <div className={className}>{children}</div>
  })
);

const sampleText = 'But I must explain to you how all this mistaken idea of denouncing \
pleasure and praising pain was born and I will give you a complete account of the system, \
and expound the actual teachings of the great explorer of the truth, \
the master-builder of human happiness.';

const tabStyles = css`padding: 24px 0 0;`;

const tabs = [
  {
    label: 'Create Replica Set',
    content: <PopupBody className={tabStyles} scrollable>
      {sampleText.repeat(10)}
    </PopupBody>
  },
  {
    label: 'Join Replica Set',
    content: <PopupBody className={tabStyles} scrollable>
      {sampleText.repeat(40)}
    </PopupBody>
  },
  {
    label: 'Bad example tab',
    content: <PopupBody className={tabStyles}>
      Bad example (without scroll).
      {sampleText.repeat(40)}
    </PopupBody>
  }
];

const action = () => null;

describe('Modal', () => {
  // afterEach(() => {
  //   ReactDOM.createPortal.mockClear()
  // })

  it('renders closed', () => {
    const tree = renderer.create(
      <Modal visible={false} onClose={action}>Modal content</Modal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders simple modal', () => {
    const tree = renderer.create(
      <Modal
        title='Simple Modal'
        visible
        onClose={action}
      >
        <PopupBody>{sampleText}</PopupBody>
        <PopupFooter
          controls={[
            <Button intent='primary' text='Accept' />,
            <Button text='Decline' />
          ]}
        />
      </Modal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders modal with large text', () => {
    const tree = renderer.create(
      <Modal
        title='Configure server'
        visible
        onClose={action}
        wide
      >
        <PopupBody>{sampleText.repeat(40)}</PopupBody>
        <PopupFooter
          controls={[
            <Button intent='primary' text='Accept' />,
            <Button text='Decline' />
          ]}
        />
      </Modal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders modal with tabs', () => {
    const tree = renderer.create(
      <Modal
        title='Configure server'
        visible
        onClose={action}
        wide
        fit
      >
        <Tabbed tabs={tabs} />
        <PopupFooter
          controls={[
            <Button intent='primary' text='Accept' />,
            <Button text='Decline' />
          ]}
        />
      </Modal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders modal with form', () => {
    const tree = renderer.create(
      <Modal
        title='Configure server'
        visible
        onClose={action}
        onSubmit={() => null}
        footerControls={[
          <Button intent='primary' text='Accept' type='submit' />,
          <Button text='Decline' onClick={action} />
        ]}
      >
        <LabeledInput label='First input' id='first-input' />
      </Modal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});