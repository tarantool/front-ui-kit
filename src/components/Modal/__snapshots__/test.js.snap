// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders closed 1`] = `null`;

exports[`Modal renders modal with form 1`] = `
<div
  className="css-fp900x"
  onMouseDown={[Function]}
>
  <form
    className="css-xynqqq"
    onKeyDown={[Function]}
    onSubmit={[Function]}
    tabIndex={0}
  >
    <h2
      className="css-34jfdi"
    >
      Configure server
    </h2>
    <svg
      className="css-1t4o2sh"
      onClick={[Function]}
      viewBox="0 0 16 16"
    >
      <use
        xlinkHref="#close"
      />
    </svg>
    <div
      className="css-i9gxme"
    >
      <label
        className="css-kf3i1p"
      >
        <div
          className="css-13kfgn1"
        >
          <span
            className="css-dpor9z"
          >
            First input
          </span>
        </div>
        <div
          className="css-142sba0"
        >
          <input
            className="css-1ssefzq"
            onBlur={[Function]}
            onFocus={[Function]}
          />
        </div>
        <p
          className="css-5ut22t"
        />
      </label>
    </div>
    <div
      className="css-vtqncq"
    >
      <div
        className="css-99wgdf"
      >
        <div
          className="css-12ibiio"
        >
          <button
            className="css-fz07pv"
            type="submit"
          >
            Accept
          </button>
        </div>
        <div
          className="css-12ibiio"
        >
          <button
            className="css-xhnfgx"
            onClick={[Function]}
            type="button"
          >
            Decline
          </button>
        </div>
      </div>
    </div>
    <div
      className="css-vnc1z7"
      onFocus={[Function]}
      tabIndex="0"
    />
  </form>
</div>
`;

exports[`Modal renders modal with large text 1`] = `
<div
  className="css-fp900x"
  onMouseDown={[Function]}
>
  <div
    className="css-8ieuwu"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <h2
      className="css-34jfdi"
    >
      Configure server
    </h2>
    <svg
      className="css-1t4o2sh"
      onClick={[Function]}
      viewBox="0 0 16 16"
    >
      <use
        xlinkHref="#close"
      />
    </svg>
    <div
      className="css-i9gxme"
    >
      <div
        className="css-1j7f3i8"
      >
        But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.
      </div>
      <div
        className="css-vtqncq"
      >
        <div
          className="css-99wgdf"
        >
          <div
            className="css-12ibiio"
          >
            <button
              className="css-fz07pv"
              type="button"
            >
              Accept
            </button>
          </div>
          <div
            className="css-12ibiio"
          >
            <button
              className="css-xhnfgx"
              type="button"
            >
              Decline
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-vnc1z7"
      onFocus={[Function]}
      tabIndex="0"
    />
  </div>
</div>
`;

exports[`Modal renders modal with tabs 1`] = `
<div
  className="css-fp900x"
  onMouseDown={[Function]}
>
  <div
    className="css-e8de1g"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <h2
      className="css-34jfdi"
    >
      Configure server
    </h2>
    <svg
      className="css-1t4o2sh"
      onClick={[Function]}
      viewBox="0 0 16 16"
    >
      <use
        xlinkHref="#close"
      />
    </svg>
    <div
      className="css-i9gxme"
    >
      <div
        className="css-1g4yje1"
      >
        <div
          className="css-0"
        >
          <button
            className="css-1x4d7qt"
            onClick={[Function]}
          >
            Create Replica Set
          </button>
          <button
            className="css-1xb0ngy"
            onClick={[Function]}
          >
            Join Replica Set
          </button>
          <button
            className="css-1xb0ngy"
            onClick={[Function]}
          >
            Bad example tab
          </button>
        </div>
        <div
          className="css-3nlq3p"
        >
          <div
            className="css-1wzifj2"
          >
            But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.
          </div>
        </div>
      </div>
      <div
        className="css-vtqncq"
      >
        <div
          className="css-99wgdf"
        >
          <div
            className="css-12ibiio"
          >
            <button
              className="css-fz07pv"
              type="button"
            >
              Accept
            </button>
          </div>
          <div
            className="css-12ibiio"
          >
            <button
              className="css-xhnfgx"
              type="button"
            >
              Decline
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-vnc1z7"
      onFocus={[Function]}
      tabIndex="0"
    />
  </div>
</div>
`;

exports[`Modal renders simple modal 1`] = `
<div
  className="css-fp900x"
  onMouseDown={[Function]}
>
  <div
    className="css-xynqqq"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <h2
      className="css-34jfdi"
    >
      Simple Modal
    </h2>
    <svg
      className="css-1t4o2sh"
      onClick={[Function]}
      viewBox="0 0 16 16"
    >
      <use
        xlinkHref="#close"
      />
    </svg>
    <div
      className="css-i9gxme"
    >
      <div
        className="css-1j7f3i8"
      >
        But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness.
      </div>
      <div
        className="css-vtqncq"
      >
        <div
          className="css-99wgdf"
        >
          <div
            className="css-12ibiio"
          >
            <button
              className="css-fz07pv"
              type="button"
            >
              Accept
            </button>
          </div>
          <div
            className="css-12ibiio"
          >
            <button
              className="css-xhnfgx"
              type="button"
            >
              Decline
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-vnc1z7"
      onFocus={[Function]}
      tabIndex="0"
    />
  </div>
</div>
`;
